bridges/otellogrus/hook.go

```diff
@@ -0,0 +1,255 @@
+		opts = append(opts, log.WithSchemaURL(c.scope.SchemaURL))
+	}
+	return c.provider.Logger(c.scope.Name, opts...)
+}
+
+// Option configures a [Hook].
+type Option interface {
+	apply(config) config
+}
+
+type optFunc func(config) config
+
+func (f optFunc) apply(c config) config { return f(c) }
+
+// WithInstrumentationScope returns an [Option] that configures the scope of
+// the logs that will be emitted by the configured [Hook].
+//
+// By default if this Option is not provided, the Hook will use a default
+// instrumentation scope describing this bridge package. It is recommended to
+// provide this so log data can be associated with its source package or
+// module.
+func WithInstrumentationScope(scope instrumentation.Scope) Option {
+	return optFunc(func(c config) config {
+		c.scope = scope
+		return c
+	})
+}
+
+// WithLoggerProvider returns an [Option] that configures [log.LoggerProvider]
+// used by a [Hook].
+//
+// By default if this Option is not provided, the Hook will use the global
+// LoggerProvider.
+func WithLoggerProvider(provider log.LoggerProvider) Option {
+	return optFunc(func(c config) config {
+		c.provider = provider
+		return c
+	})
+}
+
+// WithLevels returns an [Option] that configures the log levels that will fire
+// the configured [Hook].
+//
+// By default if this Option is not provided, the Hook will fire for all levels.
+// LoggerProvider.
+func WithLevels(l []logrus.Level) Option {
+	return optFunc(func(c config) config {
+		c.levels = l
+		return c
+	})
+}
+
+// NewHook returns a new [Hook] to be used as a [logrus.Hook].
+//
+// If [WithLoggerProvider] is not provided, the returned Hook will use the
+// global LoggerProvider.
+func NewHook(options ...Option) logrus.Hook {
+	cfg := newConfig(options)
+	return &Hook{
+		logger: cfg.logger(),
+		levels: cfg.levels,
+	}
+}
+
+// Hook is a [logrus.Hook] that sends all logging records it receives to
+// OpenTelemetry. See package documentation for how conversions are made.
+type Hook struct {
+	logger log.Logger
+	levels []logrus.Level
+}
+
+// Levels returns the list of log levels we want to be sent to OpenTelemetry.
+func (h *Hook) Levels() []logrus.Level {
+	return h.levels
+}
+
+// Fire handles the passed record, and sends it to OpenTelemetry.
+func (h *Hook) Fire(entry *logrus.Entry) error {
+	ctx := entry.Context
+	h.logger.Emit(ctx, h.convertEntry(entry))
+	return nil
+}
+
+func (h *Hook) convertEntry(e *logrus.Entry) log.Record {
+	var record log.Record
+	record.SetTimestamp(e.Time)
+	record.SetBody(log.StringValue(e.Message))
+
+	const sevOffset = logrus.Level(log.SeverityDebug) - logrus.DebugLevel
+	record.SetSeverity(log.Severity(e.Level + sevOffset))
+	record.AddAttributes(convertFields(e.Data)...)
+
+	return record
+}
+
+func convertFields(fields logrus.Fields) []log.KeyValue {
+	kvs := make([]log.KeyValue, len(fields))
+
+	i := 0
+	for k, v := range fields {
+		kvs[i] = log.KeyValue{
+			Key:   k,
+			Value: convertValue(v),
+		}
+		i++
+	}
```

Comment:
Why not `kvs := make([]log.KeyValue, 0, len(fields))` and simply appending?

```suggestion
	kvs := make([]log.KeyValue, 0, len(fields))
	for k, v := range fields {
		kvs = append(kvs, log.KeyValue{
			Key:   k,
			Value: convertValue(v),
		})
	}
```